### ISSUE TITLE
Add tracked data to our dispatchCheckoutEvent events

### ISSUE THREAD

**Issue by senadir** [woothemes/woocommerce#42424]
We currently only add the tracked event data to some of our events, but not all of them, it would be great to also add the email and address to `'set-email-address'` `'set-phone-number'` `'set-shipping-address'` and `'set-billing-address'`

**Comment by amanx123** [comment]
can you tell me more about this , I would like to work on this.

**Comment by senadir** [comment]
Sure thing @amanx123 

In the codebase, you can see several calls for `dispatchCheckoutEvent` and `dispatchStoreEvent`, some of them contain an event name and an object with extra params, others only contain the event name.

For the ones that only contain the event name, we would like add extra params like the email, phone number, and shipping or billing address. 

This should follow the same pattern that exists in other calls.

The value should be accessible near the event call.

**Comment by pierorocca** [comment]
@senadir  @ralucaStan triaging aged issues. Could you elaborate on the use cases here, why this is needed, and who this benefits?

**Comment by pierorocca** [comment]
This needs engineering input before we can confirm the final disposition.

Area to validate: This request is still unimplemented on trunk, and the desired behavior is specific enough for engineering work. The event emitters and analytics consumers are easy to identify.

@senardir there's an unanswered question  I asked a year ago. Re-reading this is there privacy/PII implications of adding those params for analytics consumers?

**Comment by senadir** [comment]
Knowing what changes in tracking is a helpful nice-to-have since you can act on those changes or report them, otherwise, those events are only useful for funnels and nothing more
